module/localisation/NUcapLocalisation/src/NUcapLocalisation.cpp

```diff
@@ -58,20 +56,21 @@ namespace localisation {
                     float x = rigidBody.position().x();
                     float y = rigidBody.position().y();
                     float z = rigidBody.position().z();
-                    UnitQuaternion q(arma::vec4{rigidBody.rotation().x(),
-                                                rigidBody.rotation().y(),
-                                                rigidBody.rotation().z(),
-                                                rigidBody.rotation().t()});
+                    Eigen::Quaternionf q(Eigen::Vector4f{rigidBody.rotation().x(),
+                                                         rigidBody.rotation().y(),
+                                                         rigidBody.rotation().z(),
+                                                         rigidBody.rotation().t()});
 
-                    Rotation3D groundToWorldRotation = q;  // * sensors.camToGround.submat(0,0,2,2).t();
 
-                    double heading = utility::math::angle::acos_clamped(groundToWorldRotation(0, 0));
+                    Eigen::Rotation<float, 3> groundToWorldRotation = q.normalized().toRotationMatrix();
```

Comment:
Make this a `Eigen::Matrix3f` and then remove the `.matrix()` calls below.
```suggestion
                    Eigen::Matrix3f groundToWorldRotation = q.normalized().toRotationMatrix();
```